### ISSUE TITLE
raw_sample in optimize_acqf

### ISSUE THREAD

**Issue by WeijingDai** [pytorch/botorch#405]
Hi, probably it is just a simple question for experienced bayesian users
What does the raw_sample indicate in optimize_acqf setting?
What is the relationship between raw_sample and num_restart/batch size q/number of available observation?
I have walked through the Botorch website and the paper introducing Botorch, but wasn't able to get the feeling of it.
Thank you very much
Weijing

**Comment by Balandat** [comment]
`raw_samples` is the number of (quasi-) random points in the heuristic that is used to generate the `num_restart` initial conditions points from which the actual gradient-based optimization is being performed. Basically, we perform a pre-optimization step by evaluating the acquisition function at `raw_samples` points, pick the best (according to some softmax so as to introduce some exploration) `num_restart` points from that, and then perform local gradient-based optimization to fine-tune the result from each of these selected points. The larger `raw_samples` the better we can cover the space and the more likely it is we start close to the global optimum (at the expense of CPU time and in particular memory usage).

**Comment by WeijingDai** [comment]
Thanks for your clear explanation. Does it mean that there is always randomness in the process of optimeze_acqf making it hardly repeat the same optimization trace in different trials?

**Comment by Balandat** [comment]
By default there is randomness, yes (this is not necessarily a bad thing - optimizing the acquisition function is a very hard non-convex problem that is often constant in much of the domain, so this randomness can be helpful in optimizing). 

If you want reproducibility, you can fix a random seed by passing `options={"seed": 1234}` to `optimize_acqf`.
